<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>JS错误处理</title>
</head>
<body>
<p>不管输入是否正确，输入框都会再输入后清空。</p>
<p>请输入 5 ~ 10 之间的数字：</p>

<input id="demo" type="text">
<button type="button" onclick="myFunction()">点我</button>
<p id="p01"></p>
<script>
    function myFunction() {
        var message, x;
        message = document.getElementById("p01");
        message.innerHTML = "";
        x = document.getElementById("demo").value;
        /*throw可以抛出一个异常，在这里抛出的是一个自定义错误*/
        try {
            if(x == "") throw "值是空的";
            if(isNaN(x)) throw "值不是一个数字";
            x = Number(x);
            if(x > 10) throw "太大";
            if(x < 5) throw "太小";
        }
        catch(err) {
            message.innerHTML = "错误: " + err;
        }
        finally {
            document.getElementById("demo").value = "";
        }
    }
</script>



<p id="demo1"></p> /*实际效果是，窗口显示在debugger处停止，需要点击以继续*/
<p>开启调试工具，在代码执行到第三行前会停止执行。</p>
<script>
    var x = 15 * 5;
    debugger;
    document.getElementById("demo1").innerHTML = x;
</script>

</body>
</html>